source/MaterialXCore/Node.cpp

```diff
@@ -498,6 +498,83 @@ void NodeGraph::setNodeDef(ConstNodeDefPtr nodeDef)
     }
 }
 
+NodeDefPtr NodeGraph::makeDeclaration(const string& nodeDefName, const string& node, string& newGraphName, const string& group)
```

Comment:
Initial creation will only create outputs since these are required to have a matching nodedef<->nodegraph signature. Inputs on the new nodedef and interfacename name tagging is the second new method.